File: PHP/connexion.php
<?php

$config = parse_ini_file('config.ini');

$dsn = "mysql:host=" . $config['host'] . ";dbname=" . $config['dbname'] . ";charset=utf8mb4";
$username = $config['username'];
$password = $config['password'];

try {
    $pdo = new PDO($dsn, $username, $password);
    $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
    $pdo->exec("set names utf8mb4");
} catch (PDOException $e) {
    http_response_code(500);
    die("Connection failed: " . $e->getMessage());
}

// type de requête
$request_method = $_SERVER['REQUEST_METHOD'];

switch ($request_method) {
    case 'POST':
        $json_data = file_get_contents('php://input');

        // décode le JSON
        $data = json_decode($json_data, true);

        // vérifier si tous les champs sont remplis
        if (empty($data['login']) || empty($data['password'])) {
            http_response_code(400); // Bad Request
            echo json_encode(array('error' => 'Veuillez remplir tous les champs'));
            exit;
        }

        $sql = "SELECT * FROM utilisateur WHERE email = :login";
        $stmt = $pdo->prepare($sql);
        $stmt->bindParam(':login', $data['login']);
        $stmt->execute();

        $user = $stmt->fetch(PDO::FETCH_ASSOC);

        if (!$user) {
            http_response_code(401);
            echo json_encode(array('error' => 'L\'adresse mail n\'existe pas.'));
            exit;
        }

        // vérifier le mot de passe
        $sqlPasswordCheck = "SELECT password FROM utilisateur WHERE email = :login";
        $stmtPasswordCheck = $pdo->prepare($sqlPasswordCheck);
        $stmtPasswordCheck->bindParam(':login', $data['login']);
        $stmtPasswordCheck->execute();

        $hashedPassword = $stmtPasswordCheck->fetchColumn();
        $id_utilisateur = $user['id_utilisateur'];
        if (password_verify($data['password'], $hashedPassword)) {

            // Connexion réussie, on log
            $sqlLog = "INSERT INTO log (message,dateheure,id_utilisateur) VALUES ('Connexion réussie', NOW(), $id_utilisateur)";
            $stmtLog = $pdo->prepare($sqlLog);
            $stmtLog->execute();

            http_response_code(200);

            $id_type = $user['id_type'];
            echo json_encode(array(
                'message' => 'Connexion réussie',
                'id_type' => $id_type,
                'id_utilisateur' => $id_utilisateur
            ));
        } else {
            $sqlLog = "INSERT INTO log (message,dateheure,id_utilisateur) VALUES ('Connexion échoué', NOW(), $id_utilisateur)";
            $stmtLog = $pdo->prepare($sqlLog);
            $stmtLog->execute();
            http_response_code(401);
            echo json_encode(array('error' => 'Identifiants incorrects'));
            exit;
        }
        break;

    case 'GET':
        if (isset($_GET['id_utilisateur'])) {
            $id_utilisateur = $_GET['id_utilisateur'];

            $sql = "SELECT utilisateur.*, fichier.chemin 
            FROM utilisateur 
            LEFT JOIN fichier ON utilisateur.id_fichier = fichier.id_fichiers 
            WHERE utilisateur.id_utilisateur = :id_utilisateur
            ";

            $stmt = $pdo->prepare($sql);
            $stmt->bindParam(':id_utilisateur', $id_utilisateur);
            $stmt->execute();

            $userInfo = $stmt->fetch(PDO::FETCH_ASSOC);

            if ($userInfo) {
                http_response_code(200);
                echo json_encode($userInfo);
            } else {
                http_response_code(404);
                echo json_encode(array('error' => 'Utilisateur non trouvé'));
            }
        } else {
            http_response_code(400);
            echo json_encode(array('error' => 'ID utilisateur manquant dans la requête'));
        }
        break;
    default:
        http_response_code(405);
        break;
}

$pdo = null;

File: lib/connexion.dart
import 'dart:convert';
import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'config.dart';
import 'technicien/accueilTechnicien.dart';
import 'politique_confidentialite.dart';

class ConnexionPage extends StatefulWidget {
  @override
  _ConnexionPageState createState() => _ConnexionPageState();
}

class _ConnexionPageState extends State<ConnexionPage> {
  String _pseudoEmailError = '';
  String _motDePasseError = '';
  bool _isObscure = true;

  final TextEditingController _loginController = TextEditingController();
  final TextEditingController _motDePasseController = TextEditingController();
  final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();

  Future<void> _submitForm() async {
    setState(() {
      _pseudoEmailError = '';
      _motDePasseError = '';
    });

    if (_loginController.text.isEmpty) {
      setState(() {
        _pseudoEmailError = 'Le champ Email est requis.';
      });
    }

    if (_motDePasseController.text.isEmpty) {
      setState(() {
        _motDePasseError = 'Le champ Mot de passe est requis.';
      });
    }

    if (_loginController.text.isEmpty || _motDePasseController.text.isEmpty) {
      _showSnackBar('Veuillez remplir tous les champs.');
      return;
    }

    Map<String, dynamic> data = {
      'login': _loginController.text,
      'password': _motDePasseController.text,
    };

    String jsonData = json.encode(data);

    final response = await http.post(
      Uri.parse('http://${AppConfig.ipAddress}/PHP/connexion.php'),
      headers: {
        'Content-Type': 'application/json',
      },
      body: jsonData,
    );

    print('Réponse de l\'API : ${response.body}');
    if (response.statusCode == 200) {
      final Map<String, dynamic> responseData = json.decode(response.body);

      if (responseData.containsKey('message')) {
        _showSnackBar(responseData['message']);

        int idType = responseData['id_type'];
        int idUser = responseData['id_utilisateur'];

        if (idType == 2) {
          Navigator.pushReplacement(
            context,
            MaterialPageRoute(
              builder: (context) => AccueilTechnicien(userId: idUser),
            ),
          );
        } else if (idType == 3) {
        } else {
          _showSnackBar('Type d\'utilisateur non géré');
        }
      } else {
        _showSnackBar('Réponse inattendue.');
      }
    } else if (response.statusCode == 401) {
      final Map<String, dynamic> responseData = json.decode(response.body);

      if (responseData.containsKey('error')) {
        if (responseData['errorpseudo'] == 'L\'adresse mail n\'existe pas.') {
          _showSnackBar('Email inexistant');
        } else if (responseData['errorpassword'] == 'Mot de passe incorrect') {
          _showSnackBar('Mot de passe incorrect');
        } else {
          _showSnackBar(responseData['error']);
        }
      } else {
        _showSnackBar('Identifiants incorrects');
      }
    }
  }

  void _showSnackBar(String message) {
    print('salut');
    final snackBar = SnackBar(
      content: Text(message),
      duration: const Duration(seconds: 3),
    );
    ScaffoldMessenger.of(context).showSnackBar(snackBar);
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      key: _scaffoldKey,
      appBar: AppBar(
        title: Text(''),
      ),
      body: SingleChildScrollView(
        child: Center(
          child: Padding(
            padding: const EdgeInsets.all(20.0),
            child: Column(
              mainAxisAlignment: MainAxisAlignment.start,
              children: [
                Container(
                  height: 150.0,
                  width: 150.0,
                  margin: const EdgeInsets.only(bottom: 20.0),
                  decoration: const BoxDecoration(
                    image: DecorationImage(
                      image: NetworkImage(
                          'http://${AppConfig.chemin}pieces_jointe/ico_mobile/logconnectservices.png'),
                      fit: BoxFit.contain,
                    ),
                  ),
                ),
                TextField(
                  controller: _loginController,
                  decoration: InputDecoration(
                    labelText: 'E-mail',
                    prefixIcon: const Icon(Icons.person),
                    errorText:
                        _pseudoEmailError.isNotEmpty ? _pseudoEmailError : null,
                  ),
                ),
                const SizedBox(height: 8),
                TextFormField(
                  controller: _motDePasseController,
                  decoration: InputDecoration(
                    labelText: 'Mot de passe',
                    prefixIcon: Icon(Icons.lock),
                    suffixIcon: IconButton(
                      icon: Icon(
                        _isObscure ? Icons.visibility : Icons.visibility_off,
                      ),
                      onPressed: () {
                        setState(() {
                          _isObscure = !_isObscure;
                        });
                      },
                    ),
                    errorText:
                        _motDePasseError.isNotEmpty ? _motDePasseError : null,
                  ),
                  obscureText: _isObscure,
                ),
                const SizedBox(height: 20),
                ElevatedButton(
                  onPressed: _submitForm,
                  child: const Text('Se connecter'),
                ),
                const SizedBox(height: 10),
                TextButton(
                  onPressed: () {
                    Navigator.push(
                      context,
                      MaterialPageRoute(builder: (context) => PrivacyPolicyPage()),
                    );
                  },
                  child: Text("Voir la politique de confidentialité"),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}